### PR TITLE
docs(blog): introducing react-view

### PR DESCRIPTION
**Pull request by tajo**
[RENDERED](https://baseweb-git-view-blog.uber-ui-platform.now.sh/blog/introducing-react-view)

### CODE REVIEW

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    SIZE,
+  }}
+  imports={{
+    'baseui/button': {
+      named: ['Button'],
+    },
+  }}
+/>
+
+All these parts work in harmony. An update one of them synchronizes the state across all of them. So you can change the knob to update the source code or edit the source code to updat knobs.
```
Comment (gergelyke) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
All these parts work in harmony. An update to one of them synchronizes the state across all of them. So you can change the knob to update the source code or edit the source code to update knobs.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+**React View is open sourced as [uber/react-view](https://github.com/uber/react-view)** and you can install with npm:
```
Comment (gergelyke) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
**React View is open sourced as [uber/react-view](https://github.com/uber/react-view)** and you can install from npm:
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+```
+yarn add react-view
+```
+
+## Wait But Why?
+
+Our team is responsible for this website and the entire Base Web component library. Naturally, you might assume that most of our jobs consists of building React components. However, as the number of Base Web users grows, the reality is slightly different:
```
Comment (gergelyke) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: Should we phrase this half-jokingly? I am a bit worried, that folks will take it too seriously

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+<BlogImage
+  src="https://user-images.githubusercontent.com/1387913/69685208-785d8d80-1070-11ea-82bf-81bd6b7f28ee.png"
+  alt="UI Platform team's workload"
+  caption="UI Platform team's workload"
+/>
+
+**And that's a good thing**. If you don’t spend a lot of time helping other developers, it probably means that nobody uses your stuff. In a single year, we went from 0 to almost 500 applications! That translates into 1000 users. There are 50 components, hundreds props and endless possibilities. It is not suprising we have to spend a lot of time with support. Is there a way to get some of that time back?
```
Comment (gergelyke) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
**And that's a good thing**. If you don’t spend a lot of time helping other developers, it probably means that nobody uses your stuff. In a single year, we went from 0 to almost 500 applications! That translates into 1000s of users. There are 50 components, hundreds of props and endless possibilities. It is not surprising we have to spend a lot of time with support. Is there a way to get some of that time back?
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+We believe that if we build an amazing documentation, we can significantly reduce the number of questions in our chats and makes developers happier. We have invested a significant time to experiment with different kinds of documentation and React View is our latest iteration.
```
Comment (gergelyke) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
We believe that if we build amazing documentation, we can significantly reduce the number of questions in our chats and makes developers happier. We have invested a significant amount of time to experiment with different kinds of documentation and React View is our latest iteration.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+
+> Build documentation so good they can’t ignore it.
+
+Does it work? Check for yourself! You can browse through our components on the right side and explore various features.
```
Comment (gergelyke) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
Does it work? Check for yourself! You can browse through our components on the left-hand side and explore various features.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+
+<LiveEditor initialCode={`2 + 5`} />
+
+## So What Really Is React View?
+
+**It is a set of tools** that aspires to close the gap between users, developers and designers of component libraries. React View aims to make documentations more interactive and useful.
+
+## Future
+
+**This is just a start for us**. We always gather the feedback and keep moving forward. There is already a [list of things](https://github.com/uber/react-view/issues/2) we want to improve. And there is also a long-term vision about how far the automatic code generation could get us.
```
Comment (gergelyke) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: should we also encourage folks to take those issues on? if we'd break it down to multiple issues, with the same label, we could still link to them and would make it easier for folks to grab only one

**Comment by gergelyke** [comment]
This is really well written 👏

**Comment by gergelyke** [comment]
nit: should we align the Read button to the bottom of the card?
Screen Shot 2019-11-27 at 9 36 39 AM

**Comment by tajo** [comment]
> This is really well written 👏

Thanks. This wasn’t meant for a review yet. I only created a draft PR to get the preview haha. Sorry about that!

**Comment by gergelyke** [comment]
> > This is really well written 👏
> 
> Thanks. This wasn’t meant for a review yet. I only created a draft PR to get the preview haha. Sorry about that!

ops, sorry about that!

File: documentation-site/pages/blog/introducing-react-view/index.mdx (hunk 3 above)
Comment (tajo) [reply documentation-site/pages/blog/introducing-react-view/index.mdx]: Yea, this should be a joke. What would make it more obvious (without just completely giving it up by appending “j/k”)? :)

File: documentation-site/pages/blog/introducing-react-view/index.mdx (hunk 7 above)
Comment (tajo) [reply documentation-site/pages/blog/introducing-react-view/index.mdx]: Yea, this was just a quick laundry list. We can probably split it and add more details later after the publishing (assuming there is enough interest)?

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+
+**It is a set of tools** that aspires to close the gap between users, developers and designers of component libraries. React View aims to make documentation more interactive and useful. It utilizes already popular tools as babel and prettier.
+
+**This is just a start for us**. We always gather the feedback and keep moving forward. There is already a [list of things](https://github.com/uber/react-view/issues/2) we want to improve. And there is also a long-term vision about how far the automatic code generation could get us.
+
+> React View is a set of tools that aspires to close the gap between users, developers and designers of component libraries.
+
+For now, we have designed React View to be as flexible as possible. Our hope is that it will resonate with the community and we will see new applications, custom built UI components and 3rd party design system integrations.
+
+**We could all save a lot of time by not writing a code that can be generated**.
```
Comment (gergelyke) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
**We could all save a lot of time by not writing code that can be generated**.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+import {BlogImage, Meta, Caption} from '../../../components/blog';
+
+import {View, PropTypes} from 'react-view';
+import {Button, SIZE} from 'baseui/button';
+
+export default Layout;
+
+<Meta data={metadata} />
+
+React View is an **interactive playground**, **documentation** and **code generator** for your components. You can use it to document your React component APIs, let users to live-edit the props and grab the generated ready-to-use source code.
```
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: let users to live-edit the props...
->
enabling users to live-edit the props...

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 10]
```diff
@@ -0,0 +1,278 @@
...
+
+The example below demonstrates all basic features of React View:
```
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: The example below demonstrates the basic features of React View

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 11]
```diff
@@ -0,0 +1,278 @@
...
+
+- The component **live preview**. (`baseui/button` in this case)
```
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: no period before opening paren

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 12]
```diff
@@ -0,0 +1,278 @@
...
+- Knobs representing `Button` props. Fully **interactive**.
```
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: Knobs representing `Button` props (fully **interactive**)

File: documentation-site/pages/blog/introducing-react-view/index.mdx (hunk 9 above)
Comment (sandgraham) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
React View is an **interactive playground**, **documentation** and **code generator** for your components. You can use it to document your React component APIs, let users live-edit props and grab the generated ready-to-use source code.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+    }}
+    imports={{
+      'baseui/button': {
+        named: ['Button'],
+      },
+    }}
+  />
+</div>
+
+All these parts work in harmony. An update to one of them synchronizes the state across all of them. So you can change a knob to update the source code or edit the source code to update knobs.
```
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: "All these parts work in harmony—an update to one of them synchronizes the state across all of them"

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+
+**React View is open sourced as [uber/react-view](https://github.com/uber/react-view)** and you can get it from npm right now:
+
+```
+yarn add react-view
+```
+
+## Wait But Why?
+
+Our team is responsible for this website and the entire Base Web component library. Naturally, you might assume that most of our jobs consists of building React components. However, as the number of Base Web users grows, some days it feels more like this:
```
Comment (sandgraham) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
Our team is responsible for this website and the entire Base Web component library. Naturally, you might assume that most of our job consists of building React components. However, as the number of Base Web users grows, some days it feels more like this:
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 15]
```diff
@@ -0,0 +1,278 @@
...
+
+<BlogImage
+  src="https://user-images.githubusercontent.com/1387913/69685208-785d8d80-1070-11ea-82bf-81bd6b7f28ee.png"
+  alt="UI Platform team's workload"
+  caption="UI Platform team's workload"
+  style={{maxWidth: '400px'}}
+/>
+
+**And that's a good thing**. If you don’t spend a lot of time helping other developers, it means that nobody uses your tools. We went from 0 to almost 500 applications at Uber in a single year! That translates into about 1000 of developers. There are 50 components, hundreds of props and endless possibilities. Is there a way to reduce the burden of support?
```
Comment (sandgraham) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
**And that's a good thing**. If you don’t spend a lot of time helping other developers, it means that nobody uses your tools. We went from 0 to 500 applications at Uber in a single year! That translates into about 1000 developers. There are over 50 components, hundreds of props and endless possibilities. Is there a way to reduce the burden of support?
```
Ran analytics this morning - over 500 projects now 😀.

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 16]
```diff
@@ -0,0 +1,278 @@
...
+
+All these parts work in harmony. An update to one of them synchronizes the state across all of them. So you can change a knob to update the source code or edit the source code to update knobs.
+
+**React View is open sourced as [uber/react-view](https://github.com/uber/react-view)** and you can get it from npm right now:
+
+```
+yarn add react-view
+```
+
+## Wait But Why?
```
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: Wait...But Why?

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 17]
```diff
@@ -0,0 +1,278 @@
...
+<BlogImage
+  src="https://user-images.githubusercontent.com/1387913/69685208-785d8d80-1070-11ea-82bf-81bd6b7f28ee.png"
+  alt="UI Platform team's workload"
+  caption="UI Platform team's workload"
+  style={{maxWidth: '400px'}}
+/>
+
+**And that's a good thing**. If you don’t spend a lot of time helping other developers, it means that nobody uses your tools. We went from 0 to almost 500 applications at Uber in a single year! That translates into about 1000 of developers. There are 50 components, hundreds of props and endless possibilities. Is there a way to reduce the burden of support?
+
+We believe that by building na amazing documentation we can significantly reduce the number of questions in our chats and make developers happier. We have invested a significant amount of time to experiment with different kinds of documentation and React View is our latest iteration.
```
Comment (sandgraham) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
We believe that by building amazing documentation we can significantly reduce the number of questions we receive in support channels and make developers happier. We have invested a significant amount of time to experiment with different kinds of documentation and React View is our latest iteration.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx (hunk 14 above)
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: Drop "naturally"

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 18]
```diff
@@ -0,0 +1,278 @@
...
+> Build documentation so good they can’t ignore it.
+
+Does it work? Check for yourself! You can browse through our components on the left-hand side and explore various features.
+
+## How to Use React View
+
+**There are three main ways** of using React View:
+
+- **All-in-one playground**. Import `<View />`, give it the component configuration and drop it in your web documentation. This is ideal if you want to start as quickly as possible and don't need to customize anything.
+- **Build your own playground**. Import the `useView` hook and give it the component configuration. This hook handles the playground state and returns various props and callbacks that you can fit into your own UI components. React View also exports all default UI parts separately (`Editor`, `ActionButtons`, `Compiler`, `Knobs`, `Error`...) so you can reuse them. This is a great option if you want to customize some parts of the default UI (or all of it) and not to worry about the rest.
```
Comment (sandgraham) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
- **Build your own playground**. Import the `useView` hook and give it the component configuration. This hook handles the playground state and returns various props and callbacks that you can fit into your own UI components. React View also exports all default UI parts separately (`Editor`, `ActionButtons`, `Compiler`, `Knobs`, `Error`...) so you can reuse them. This is a great option if you want to customize some parts of the default UI (or all of it) and not worry about the rest.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx (hunk 17 above)
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: building na amazing -> building amazing

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 19]
```diff
@@ -0,0 +1,278 @@
...
+      <Error {...params.errorProps} />
+      <ActionButtons {...params.actions} />
+    </React.Fragment>
+  );
+};
+```
+
+### Live Code Editing Only
+
+If you don't need the code generation or props documentation, you can also use React View as an editable source code with the live preview.
```
Comment (sandgraham) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
If you don't need the code generation or props documentation, you can also use React View for editing source code with a live preview.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 20]
```diff
@@ -0,0 +1,278 @@
...
+
+<LiveEditor initialCode={`<h3>Hey</h3>`} />
+
+Or pretty much anything that can be executed after the return statement of JavaScript function:
+
+<LiveEditor initialCode={`2 + 5`} />
+
+## So What Really Is React View?
+
+**It is a set of tools** that aspires to close the gap between users, developers and designers of component libraries. React View aims to make documentation more interactive and useful. It utilizes already popular tools as babel and prettier.
```
Comment (sandgraham) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
**It is a set of tools** that aspires to close the gap between users, developers and designers of component libraries. React View aims to make documentation more interactive and useful. It utilizes already popular tools such as babel and prettier.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 21]
```diff
@@ -0,0 +1,278 @@
...
+
+**This is just a start for us**. We always gather the feedback and keep moving forward. There is already a [list of things](https://github.com/uber/react-view/issues/2) we want to improve. And there is also a long-term vision about how far the automatic code generation could get us.
```
Comment (sandgraham) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]:
```suggestion
**This is just a start for us**. We are always gathering feedback and moving forward. There is already a [list of things](https://github.com/uber/react-view/issues/2) we want to improve. And there is also a long-term vision about how far the automatic code generation could get us.
```

File: documentation-site/pages/blog/introducing-react-view/index.mdx (hunk 18 above)
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: that you can fit into your own UI components -> that you can use in your own UI components
and not to worry about the rest -> without worrying about the rest

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 22]
```diff
@@ -0,0 +1,278 @@
...
+
+## How to Use React View
+
+**There are three main ways** of using React View:
+
+- **All-in-one playground**. Import `<View />`, give it the component configuration and drop it in your web documentation. This is ideal if you want to start as quickly as possible and don't need to customize anything.
+- **Build your own playground**. Import the `useView` hook and give it the component configuration. This hook handles the playground state and returns various props and callbacks that you can fit into your own UI components. React View also exports all default UI parts separately (`Editor`, `ActionButtons`, `Compiler`, `Knobs`, `Error`...) so you can reuse them. This is a great option if you want to customize some parts of the default UI (or all of it) and not to worry about the rest.
+- **Live code editing only**. Sometimes it is useful to only have an editable source code and live preview without the list of props. You can use `useView` for that too. Just don't give it any component configuration and don't render the `<Knobs />` component. React View uses babel, so **you can add additional presets and enable TypeScript**.
```
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: it is -> it's
an editable -> editable

File: documentation-site/pages/blog/introducing-react-view/index.mdx [hunk 23]
```diff
@@ -0,0 +1,278 @@
...
+    }}
+  />
+);
+```
+
+This is a basic example demonstrating the all-in-one `View` component. You need to define the component name, props, scope and imports. This example renders the demo at the beginning of this article (except the `size` prop).
+
+### Build Your Own Playground
+
+**Bring your own UI**. The View component is a tiny wrapper around the `useView` hook. If you wish to customize and have more control, you can use the `useView` hook directly. You can still re-use our default UI components or create your own:
```
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: wish -> want

File: documentation-site/pages/blog/introducing-react-view/index.mdx (hunk 20 above)
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: this paragraph would make a good intro

File: documentation-site/pages/blog/introducing-react-view/index.mdx (hunk 21 above)
Comment (angus-c) [inline documentation-site/pages/blog/introducing-react-view/index.mdx]: We always gather the feedback -> We're always gathering feedback

**Pull request merged**